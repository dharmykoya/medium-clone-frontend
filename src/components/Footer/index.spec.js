import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { Footer } from './index.jsx';

const prop = {
  theme: 'light-theme'
};

describe('Render component', () => {
  it('should render correctly in "debug" mode', () => {
    const component = mount(
      <BrowserRouter>
        <Footer theme={prop} />
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
    expect(component.contains(<h5>Connect with Us</h5>)).toBeTruthy();
  });

  it('should have Icon rendered', () => {
    const component = mount(
      <BrowserRouter>
        <Footer theme={prop} />
      </BrowserRouter>
    );
    expect(
      component.contains(
        <img src="./../src/assets/images/google-plus.svg" alt="google plus" />
      )
    ).toBeTruthy();
  });

  it('should have Icon rendered', () => {
    const component = mount(
      <BrowserRouter>
        <Footer theme={{ theme: 'dark-theme' }} />
      </BrowserRouter>
    );
    expect(
      component.contains(
        <img
          src="./../src/assets/images/google-plus-light.svg"
          alt="google plus light"
        />
      )
    ).toBeTruthy();
    expect(
      component.contains(
        <img
          src="./../src/assets/images/facebook-light.svg"
          alt="facebook light"
        />
      )
    ).toBeTruthy();
  });

  it('should have Icon rendered', () => {
    const component = mount(
      <BrowserRouter>
        <Footer theme={{ theme: 'dark-theme' }} />
      </BrowserRouter>
    );

    expect(
      component.contains(
        <img
          src="./../src/assets/images/facebook-light.svg"
          alt="facebook light"
        />
      )
    ).toBeTruthy();
  });
});
