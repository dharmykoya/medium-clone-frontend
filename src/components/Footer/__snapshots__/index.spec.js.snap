// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component should render correctly in "debug" mode 1`] = `ReactWrapper {}`;
